server/src/main/java/org/apache/iotdb/db/writelog/node/ExclusiveWriteLogNode.java

```diff
@@ -115,8 +113,15 @@ public void write(PhysicalPlan plan) throws IOException {
     } catch (BufferOverflowException e) {
       // if the size of a single plan bigger than logBufferWorking
       // we need to clear the buffer to drop something wrong that has written.
+      DataOutputStream dos = new DataOutputStream(new ByteArrayOutputStream());
+      plan.serialize(dos);
+      int neededSize = dos.size();
+      dos.close();
```

Comment:
It seems that there is no need to really do the serialization to get the actual size, maybe you can add a new method like `getSerializedSize()` in `PhysicalPlan`, each actual plan class to override that method to calculate its own size. In this way, you can avoid allocate extra memory in this `BufferOverflowException` situation.

Reply:
> It seems that there is no need to really do the serialization to get the actual size, maybe you can add a new method like `getSerializedSize()` in `PhysicalPlan`, each actual plan class to override that method to calculate its own size. In this way, you can avoid allocate extra memory in this `BufferOverflowException` situation.

Yes, I totally agree with what you said. In fact, I wanted to do the same thing at first. 

However, I noticed that there are quite a lot of classes extends `PhysicalPlan`. It needs to modify code massively. Considering that `getSerializedSize()` is only used for this exception and exception may be occurred rarely, I allocate extra memory to get the actual size. 

Do you think it's necessary to do what you said? If so, I may need more time to implement `getSerializedSize()` in each actual plan class.